models/azure/workflow.go

```diff
@@ -0,0 +1,59 @@
+// Copyright (C) 2022 Specter Ops, Inc.
+//
+// This file is part of AzureHound.
+//
+// AzureHound is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// AzureHound is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+package azure
+
+import "strings"
+
+type Workflow struct {
```

Comment:
Regarding workflows, one property of it is `definition`, which returns the actual workflow with actions and triggers. This could become rather verbose quickly and I'm not sure if this is wanted with respect to the Bloodhound import?

Reply:
One of the things that has become kind of painful as we try to better support Azure in BloodHound is the lack of information that we're able to capture and use that is actually relevant. So while it's awfully noisy, capturing that extra data from the start has a lower maintenance cost than having to go back and update the client every time we learn something new.

Reply:
Regarding the definition, I've implemented the generic objects as it sits in the MSDN documentation. For actions and triggers however, I don't believe it feasible to start supporting every type with it's custom elements(see image below). I recommend instead to extract them as data(`interface{}`) and either defer it to Bloodhound or add support along the way for common occurrences.

![image](https://user-images.githubusercontent.com/10261812/190409931-eb693379-18be-44d8-bf22-adf2cb607baf.png)